### ISSUE TITLE
Is there any way to have JSPM transpile module at time of install?

### ISSUE THREAD

**Issue by DavidKDeutsch** [jspm/jspm-cli#2379]
I was wondering if there was a package override that would cause JSPM to make ES6 modules "SystemJS friendly" when `jspm install` is run, as opposed to when bundling or in the browser? That way, overrides for ES6 modules could be published to the JSPM registry without people having to know which transpilers to install and whatnot. As far as I can tell, you can do this at bundle time, but I'm not sure if it is doable upon install.

**Comment by ArmorDarks** [comment]
Well, idea to transpile them _right_ after `jspm install` seems to be reasonable. It will significantly reduce initial loading time.

**Comment by aluanhaddad** [comment]
This sounds like an interesting idea but I'm not entirely clear on what you have in mind.
Do you a: want _anonymous_ `System.register` modules output for each module in the installed package written to disk? Or b: a single bundle containing _named_ `System.register` output comprising the entire package but not its dependencies?

The module format is detected on installation and this already saves a fair amount of time.

Also, if they are transpiled ahead of time, then features the like the Rollup optimization performed by `$ jspm build` would not be available.

**Comment by ArmorDarks** [comment]
> Also, if they are transpiled ahead of time, then features like the Rollup optimization performed by $ jspm build would not be available.

Well, that can be fixed by keeping original and transpiled versions on disk. Forcing saving transpiled version can be hidden behind the flag.

**Comment by DavidKDeutsch** [comment]
@aluanhaddad, I was thinking more along the lines of the former, though it wouldn't necessarily have to output a SystemJS module; it could also output, say a CommonJS module (since these work with SystemJS "out of the box", and would also address the Rollup scenario you mentioned). 

FWIW, the use case that prompted me to ask the question is a third party package which contains both an ES6 and a UMD version of its code, with the UMD version being (until recently) incorrectly generated, and thus unusable. It appears to be the case that the UMD version was created explicitly for SystemJS users, and due to the problems related to that file the author says that SystemJS is "not supported" because it caused too many issues. Since it is an Angular component, my guess is that most of its consumers use Webpack or some other system that uses the correct, ES6 version. 

So while it is true that people can use it with SystemJS, they have to figure out that they need to use a transpiler, figure out whether they transpile during bundling and/or at run-time, etc. I figure that in the same spirit as config overrides in the JSPM registry, it would instead be better for the changes needed to make a package work "out of the box" were moved upstream.

Actually, now that I think about it, having an arbitrary script that could run on install would accomplish this (and any other case), so I just now did a bit of googling and... it looks like that functionality was indeed introduced almost exactly a year ago in #2121. I'll give it a whirl and see if it accommodates my use case...
